src/core/thread/key_manager.hpp

```diff
@@ -598,6 +607,10 @@ class KeyManager : public InstanceLocator, private NonCopyable
     Mle::KeyMaterial mMleKey;
     Mle::KeyMaterial mTemporaryMleKey;
 
+#if OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
```

Comment:
For future use case consideration, how about we rename this macro to make it more generic?
```suggestion
#if OPENTHREAD_CONFIG_WAKEUP_DEVICE_ENABLE
```

Reply:
I think there's discussions about allowing many-to-many connections after wake up, which may not be parent-child topology, but more symmetric relation ship.

Reply:
Not sure about the change, for now the behavior won't be listening for wake-up frames only, but everything related to WED as defined in the spec, like MLE procedures, CSL/CST handling, etc. I would keep it as is for now.

Also this was already introduced in a previous PR, should be handled out of the scope of this PR.

Reply:
As we may add new API(`otError otLinkSetWakeUpListenCallback(otInstance *aInstance, otWakeUpCallback aCallback);`) to change the wake up behavior. I think we should leave API name focus on its behavior. Do you think we can rename `otLinkSetWedListenInterval` to `otLinkSetWakeUpListenInterval`, `otLinkSetWedListenDuration` to `otLinkSetWakeUpListenDuration`?

I'm fine with dealing with the macro name in future PRs.

Reply:
I see your concern, but those are actually names from the spec: `WED_LISTEN_DURATION` and `WED_LISTEN_INTERVAL`. I would prefer to keep them as it is for better adhesion to the spec.

Reply:
IIUC, the spec is not finalized yet. Perhaps we can continue the discussion in the spec about the configuration name. Actually the CLI is also using `wakeup` instead of `wed`. To unblock this PR and keep forward compatible, perhaps we can combine the wake up parameters as `otLinkSetWakeUpListenParameters` and `otLinkGetWakeUpListenParameters`. In the CLI, we can use `wakeup set <interval> <duration>`. As I mentioned in the other comment, this also allows us to validate the parameters at once. What do you think?

Reply:
Okay, let's do as you suggest. I'll only change the CLI command to `wakeup parameters <interval> <duration>` for better clarity.

Reply:
thank you